src/main/java/user_use_cases/RegisterUserInteractor.java

```diff
@@ -0,0 +1,43 @@
+package user_use_cases;
+
+import entities.User;
+import entities.UserFactory;
+import user_screens.RegisterUserPresenter;
+
+public class RegisterUserInteractor{
+
+    private final UserFactory factory;
+    private final UserDatabaseGateway gateway;
+    private final RegisterUserPresenter presenter;
+
+    public RegisterUserInteractor(UserFactory factory, UserDatabaseGateway gateway,
+                                  RegisterUserPresenter presenter) {
+        this.factory = factory;
+        this.gateway = gateway;
+        this.presenter = presenter;
+    }
+
+    public RegisterUserResponseModel CreateUser(RegisterUserRequestModel requestModel) {
+
+        if (!gateway.userExists(requestModel.getUsername())) {
+            return presenter.prepareRegisterFailView("Username is Taken");
```

Comment:
Should be the negation, if gateway.userExists then that means the username is taken, if the user does not exist then that means the username can be used